Summary: Filter out samples tagged with group: setup/teardown

joanlopez opened the pull request:
## What?

It filters out samples tagged with `group: ::setup` or `group: ::teardown` so they aren't collected as "group samples".

## Why?

Since #4089, we modified not only how the data in the summary is displayed but also how it is collected. So, since then, we started to collect samples with `group: ::setup` or `group: ::teardown`, while previously we did not because there are no groups named like `::setup` or `::teardown`, we only "artificially" tag samples with the `group` label (more details [here](https://grafana.com/docs/k6/latest/using-k6/workaround-iteration-duration/)).

So, for now, and until we decide the opposite (if any), we want to preserve the old behavior, where none of these two are displayed in the summary as "real" groups.

## Checklist

<!-- 
If you haven't read the contributing guidelines https://github.com/grafana/k6/blob/master/CONTRIBUTING.md 
and code of conduct https://github.com/grafana/k6/blob/master/CODE_OF_CONDUCT.md yet, please do so
-->

- [X] I have performed a self-review of my code.
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [X] I have added tests for my changes.
- [X] I have run linter and tests locally (`make check`) and all pass.

## Checklist: Documentation (only for k6 maintainers and if relevant)

**Please do not merge this PR until the following items are filled out.**

- [ ] I have added the correct milestone and labels to the PR.
- [ ] I have updated the release notes: _link_
- [ ] I have updated or added an issue to the [k6-documentation](https://github.com/grafana/k6-docs): grafana/k6-docs#NUMBER if applicable
- [ ] I have updated or added an issue to the [TypeScript definitions](https://github.com/grafana/k6-DefinitelyTyped/tree/master/types/k6): grafana/k6-DefinitelyTyped#NUMBER if applicable

<!-- - [ ] Any other relevant item -->

## Related PR(s)/Issue(s)

Contribute to #4724

codebien reviewed (commented):
LGTM. However, we might see CPU spikes for tests with high usage of groups. So, it could require to rethink a bit the end-to-end flow for this specific case. Something to follow-up in the future.

joanlopez commented:
> LGTM. However, we might see CPU spikes for tests with high usage of groups. So, it could require to rethink a bit the end-to-end flow for this specific case. Something to follow-up in the future.

My (perhaps wrong) feeling tells me that it's improbable that this comparison will ever have a significant impact (in the form of CPU spikes), because I think the operations around the new code (e.g. the `strings.Split` right after) are significantly slower. But yeah, if that ever happens, we can explore ways to optimize this. For instance, we could "cache" the result of certain operations.

EDIT: In any case, I replaced the `strings.Contains` with straight equality comparison against constant values, which should be a bit faster, if the previous code wasn't already optimized by the compiler, which I would not be surprised by, if so.

oleiade reviewed (approved):
🚀
